blog/2022/2022-04-03-new-home-for-the-loopback-blog.md

```diff
@@ -0,0 +1,115 @@
+stack - Jekyll. Although it worked fine and integrated well with GitHub Pages,
+many of the maintainers were not well-versed in the world of Ruby. This was
+compounded by shell scripts that pulled data from other Git Repositories and
+modified the site as part of the build process. Hence, a lot of time was spent
+trying to understand how the website was put together and figuring out the best
+way to tweak it to integrate the new blog.
+
+The "Strong Blog" was also written with Jekyll and hosted on GitHub Pages. Hence
+lifting and shifting was a real possibility that we contemplated. However,
+the slow build times and the maintainers' unfamiliarity with Ruby and Jekyll
+meant that we realised that the website's current stack wasn't worth keeping
+around in the long run. At the same time, performing a full migration of the
+website, documentation, and blog in one go was too daunting of a task for the
+maintainers who were already working on the project on their own dime.
+
+The idea of using a third-party service such as [Medium](https://www.medium.com)
+was also considered. However utlimately, the lack of control over the interface,
+the inability to place it in the same domain name as the main website and
+documentation, and the frustration that stem from a reader being forced to use
+a third-party service meant that it was quickly off the table.
+
+As a result, we decided to come to a compromise.
+
+## Picking A Blogging Stack
+
+After some deliberation we decided to create a hybrid stack which left the
+website and documentation with Jekyll, and use
+[Docosaurus 2](https://docusaurus.io) as the new blogging platform. We found
+Docusaurus to be a good fit as it was a Facebook-maintained open source project
+which was written on top of React. This JavaScript-based stack meant that there
+was more opportunity for extensibility and a reduced barrier for contributing
+and tweaking. It also had the benefit of speedy build times! The first-class
+support for MDX was also a plus as it meant we were able to copy over most of
+our posts with only a few tweaks, and the ability to generate the blog as build
+artifacts made it compatible with the Jamstack-eque system of GitHub Pages.
+
+Thanks to the amazing work done by [Diana](https://github.com/dhmlau), all of
+the LoopBack 4 blog posts from 2019 and 2020 were migrated to Docusaurus. With
+her work, we were able to quickly build and see the new blog coming together
+with all of the past blog posts populating the blog.
+
+## Merging Jekyll and Docusaurus
+
+When we started building the blog, we hosted it on GitHub Pages in its own Git
+Repository and served it in its own sub-domain. This allowed us to quickly build
+and patch any bugs that may have cropped up. However the end-goal was to host it
+under the same domain as the main website and documentation:
+https://loopback.io/blog. Hence, we had to create a solution which merged the
```

Comment:
I forgot whether Docusaurus will make it as a hyperlink. If not, would be good to use:
```
[https://loopback.io/blog](https://loopback.io/blog)
```